### PR TITLE
Record dot rewrite

### PR DESCRIPTION
**Pull request by Creepsy**
The entire codebase now uses the Language Extension "OverloadedRecordDots" for record field access. This change got proposed because of many conflicting field names.

### CODE REVIEW

File: src/User.hs [hunk 1]
```diff
@@ -35,13 +38,13 @@ data DbUser = DbUser {userId :: Int, username :: Text, passwordHash :: Text} der
...
 
 instance FromJSON RegisterRequest where
   parseJSON (Object o) = RegisterRequest <$> o .: "username" <*> o .: "password"
   parseJSON other = prependFailure "Parsing Register Request failed: " (typeMismatch "Object" other)
 
-data LoginRequest = LoginRequest {loginUsername :: Text, loginPassword :: Text}
+data LoginRequest = LoginRequest {username :: Text, password :: Text}
 
 instance FromJSON LoginRequest where
   parseJSON (Object o) = LoginRequest <$> o .: "username" <*> o .: "password"
```
Comment (rehwinkel) [inline src/User.hs]: Wenn username nun anders heißt, können wir FromJSON deriven

File: src/User.hs [hunk 2]
```diff
@@ -35,13 +38,13 @@ data DbUser = DbUser {userId :: Int, username :: Text, passwordHash :: Text} der
 
 instance FromRow DbUser
 
-data RegisterRequest = RegisterRequest {registerUsername :: Text, registerPassword :: Text}
+data RegisterRequest = RegisterRequest {username :: Text, password :: Text}
 
 instance FromJSON RegisterRequest where
   parseJSON (Object o) = RegisterRequest <$> o .: "username" <*> o .: "password"
   parseJSON other = prependFailure "Parsing Register Request failed: " (typeMismatch "Object" other)
```
Comment (rehwinkel) [inline src/User.hs]: FromJSON deriven

File: src/User.hs [hunk 3]
```diff
@@ -1,6 +1,9 @@
 {-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE ImportQualifiedPost #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DuplicateRecordFields #-}
```
Comment (rehwinkel) [inline src/User.hs:4]: DuplicateRecordFields mit NoFieldSelectors noch nötig?

**Review by rehwinkel — CHANGES_REQUESTED** [review]

File: src/User.hs (hunk 3 above)
Comment (Creepsy) [reply src/User.hs:4]: Yes, `DuplicateRecordFields` is still necessary even though we are using `NoFieldSelectors`.

File: src/User.hs (hunk 1 above)
Comment (Creepsy) [reply src/User.hs]: Resolved by 6c35a64ddba4e6f7781007da4a0c50d83bfc964e

File: src/User.hs (hunk 2 above)
Comment (Creepsy) [reply src/User.hs]: Resolved by 6c35a64ddba4e6f7781007da4a0c50d83bfc964e

**Pull request merged**